pkg/v1/partial/configlayer_test.go

```diff
@@ -0,0 +1,97 @@
+package partial
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+type testUIC struct {
+	configFile []byte
+}
+
+func (t testUIC) RawConfigFile() ([]byte, error) {
+	return t.configFile, nil
+}
+
+func (t testUIC) MediaType() (types.MediaType, error) {
+	return "", nil
+}
+
+func (t testUIC) LayerByDiffID(h v1.Hash) (UncompressedLayer, error) {
+	return nil, fmt.Errorf("no layer by diff ID %v", h)
+}
+
+func TestUncompressedLayerByDigest(t *testing.T) {
+	uic := testUIC{
+		configFile: []byte("{}"),
+	}
+	uie := &uncompressedImageExtender{
+		UncompressedImageCore: uic,
+	}
+	testLayerByDigestForConfigHash(t, uie)
+}
+
+type testCIC struct {
+	configFile []byte
+}
+
+func (t testCIC) RawConfigFile() ([]byte, error) {
+	return t.configFile, nil
+}
+
+func (t testCIC) MediaType() (types.MediaType, error) {
```

Comment:
If these methods are not called during the test, you can simply embed the `v1.Image` interface into `testCIC`/`testUIC` to have the test structs implement the interface. If the methods are ever called, they'll panic, but it helps reduce no-op code and prevent future breakages if you can get away with it.

```
type testCIC struct {
  v1.Image
  configFile []byte
}

func (t testCIC) RawConfigFile() ([]byte, error) { return t.configFile, nil }
```

Reply:
Nice. I'm keeping `testCIC.LayerByDigest` around because otherwise it panics. I'll remove the rest.